### DISCUSSION TITLE
execute functions whilst in debug mode

### DISCUSSION THREAD

**Discussion by ghost** [godotengine/godot-proposals#6171] [Engine Core]
there should be a little bar kind of like the line edit node at the bottom of the output tab when you test your project in debug mode and this would allow you to trigger functions e.g
---------------------this part is in the script-----------------
`func my_function(var_1,var_2):
      print(var_1 + var_2)`
-----------------this part is in the output tab---------------
and you type into the bar:
`my_function("hello ","world")` 

result:
`hello world`

**Comment by filantus** [comment]
Do you mean [REPL](https://en.wikipedia.org/wiki/Read%E2%80%93eval%E2%80%93print_loop) ?
There is REPL plugin for Godot 3 - https://godotengine.org/asset-library/asset/857.
But unfortunately didn`t see something same for Godot 4 yet.

**Reply by Zireael07** [reply]
There is a PR implementing REPL for Godot 4, but it wasn't merged yet

**Reply by Calinou** [reply]
See https://github.com/godotengine/godot/pull/60134.

**Reply by Zireael07** [reply]
thanks, search-fu failed me
